Sources/Amplitude/AMPStorage.m

```diff
@@ -0,0 +1,107 @@
+//
+//  File.m
+//
+//
+//  Created by Dante Tam on 6/10/21.
+//
+
+#import <Foundation/Foundation.h>
+
+#import "AMPStorage.h"
+
+@interface AMPStorage ()
+@end
+
+@implementation AMPStorage {
+    
+}
+
++ (NSString *)getAppStorageAmpDir {
+    NSArray *paths = NSSearchPathForDirectoriesInDomains(NSApplicationSupportDirectory, NSUserDomainMask, YES);
+    NSString *path = [paths firstObject];
+    NSString *bundleIdentifier = [[NSBundle mainBundle] bundleIdentifier];
+    return [NSString stringWithFormat:@"%@/%@/%@", path, @"/", bundleIdentifier];
+}
+
++ (NSString *)getDefaultEventsFile {
+    NSString *baseDir = [AMPStorage getAppStorageAmpDir];
+    NSString *path = [baseDir stringByAppendingString:@"/amplitude_event_storage.txt"];
+    return path;
+}
+
++ (NSString *)getDefaultIdentifyFile {
+    NSString *baseDir = [AMPStorage getAppStorageAmpDir];
+    NSString *path = [baseDir stringByAppendingString:@"/amplitude_identify_storage.txt"];
+    return path;
+}
+
++ (void)storeEvent:(NSString *) event {
+    NSString *path = [AMPStorage getDefaultEventsFile];
+    NSURL *url = [NSURL fileURLWithPath:path];
+    [AMPStorage storeEventAtUrl:url event:event];
+}
+
++ (void)storeIdentify:(NSString *) event {
+    NSString *path = [AMPStorage getDefaultIdentifyFile];
+    NSURL *url = [NSURL fileURLWithPath:path];
+    [AMPStorage storeEventAtUrl:url event:event];
+}
+
++ (void)storeEventAtUrl:(NSURL *)url event:(NSString *)event {
+    NSFileManager *fm = [NSFileManager defaultManager];
+    NSString *path = [AMPStorage getDefaultEventsFile];
+    
+    bool newFile = false;
+    if (![fm fileExistsAtPath:path]) {
+        [AMPStorage start:path];
+        newFile = true;
+    }
+    
+    NSData *jsonData = [event dataUsingEncoding:NSUTF8StringEncoding];
+    NSFileHandle *handle = [NSFileHandle fileHandleForWritingToURL:url error:nil];
+    [handle seekToEndOfFile];
+    if (!newFile) {
+        [handle writeData:[@"," dataUsingEncoding:NSUTF8StringEncoding]];
+    }
+    [handle writeData:jsonData];
+    //[handle closeFile];
+}
+
++ (void)start:(NSString *)path {
+    NSString *contents = @"{ \"batch\": [";
```

Comment:
What is the `batch` for?

Reply:
It's for keeping the JSON object structured so that it's not just an array. There may be other data associated with this file. The original use in Segment API is that they also added the upload date as a second key-value property in this file.

Reply:
This is fine. Resolved.